ui/src/services/adb-key-service.ts

```diff
@@ -0,0 +1,106 @@
+import { injectable } from 'inversify'
+import { makeAutoObservable } from 'mobx'
+
+import { socket } from '@/api/socket'
+import { AdbKeysChangedMessage } from '@/types/adb-keys-changed-message.type'
+import { UserResponseUser, UserResponseUserAdbKeysItem } from '@/generated/types'
+
+import { queries } from '@/config/queries/query-key-store'
+import { queryClient } from '@/config/queries/query-client'
+
+@injectable()
+export class AdbKeyService {
+  private adbKeyToDelete: UserResponseUserAdbKeysItem = { fingerprint: '', title: '' }
+
+  deviceKey = ''
+  deviceTitle = ''
+  errorMessage = ''
+  isAddAdbKeyOpen = false
+
+  constructor() {
+    makeAutoObservable(this)
+
+    this.onAdbKeysAdded = this.onAdbKeysAdded.bind(this)
+    this.onAdbKeysRemoved = this.onAdbKeysRemoved.bind(this)
+    this.onAdbKeysError = this.onAdbKeysError.bind(this)
+  }
+
+  private extractHostNameFromKey(key: string): string {
+    if (key.match(/.+= (.+)/)) {
+      return key.replace(/.+= (.+)/g, '$1')
+    }
```

Comment:
A better way to extract it would be to split by spaces and take everything after the first space
```suggestion
    if (key.match(/\S+\s+(.+)/)) {
      return key.replace(/\S+\s+(.+)/g, '$1')
    }
```